src/ilabs_streamsync/streamsync.py

```diff
@@ -9,39 +20,104 @@ class StreamSync:
+        return (srate, wav_signal[:,channel], wav_signal[:,1-channel])
+
     def do_syncing(self):
         """Synchronize all streams with the reference stream."""
         # TODO (waves hands) do the hard part.
         # TODO spit out a report of correlation/association between all pairs of streams
-        pass
 
-    def plot_sync(self):
-        pass
+    def plot_sync_pulses(self, tmin=0, tmax=float('inf')):
+        """Plot each stream in the class."""
+        # TODO Plot the raw file on the first plot.
+        fig, axset = plt.subplots(len(self.streams)+1, 1, figsize = [8,6]) #show individual channels seperately, and the 0th plot is the combination of these. 
+        for i, stream in enumerate(self.streams):
+            npts = len(stream[2])
+            tt = np.arange(npts) / stream[1]
+            idx = np.where((tt>=tmin) & (tt<tmax))
+            axset[i+1].plot(tt[idx], stream[2][idx].T)
+            axset[i+1].set_title(pathlib.Path(stream[0]).name)
+            # Make label equal to simply the cam number
+        plt.show()
+
+def extract_audio_from_video(path_to_video, output_dir):
+    """Extract audio from path provided.
+
+    path_to_video: str
+        Path to audio file
+        TODO allow path_to_video to take regex?
+    output_dir: str
+        Path to directory where extracted audio should be sent
+
+    Effects:
+        Creates output directory if non-existent. For each video found, creates
+        a file with the associated audio labeled the same way.
+
+    Raises:
+        ValueException if video path does not exist, 
+        Exception if filename is taken in output_dir
+    """
+    audio_codecout = 'pcm_s16le'
+    audio_suffix = '_16bit'
+    p = pathlib.Path(path_to_video)
+    audio_file = p.stem + audio_suffix + '.wav'
+    if not p.exists():
+        raise ValueError('Path provided cannot be found.')
+    if pathlib.PurePath.joinpath(pathlib.Path(output_dir), pathlib.Path(audio_file)).exists():
+        raise Exception(f"Audio already exists for {path_to_video} in output directory.")
 
+    command = ['ffmpeg',
+        '-acodec', 'pcm_s24le',       # force little-endian format (req'd for Linux)
+        '-i', path_to_video,
+        '-map', '0:a',                # audio only (per DM)
+#         '-af', 'highpass=f=0.1',
+        '-acodec', audio_codecout,
+        '-ac', '2',                   # no longer mono output, so setting to "2"
+        '-y', '-vn',                  # overwrite output file without asking; no video
+        '-loglevel', 'error',
+        audio_file]
+    pipe = subprocess.run(command, timeout=50, check=False)
 
-def extract_audio_from_video(path_to_video, channel):
-    """Path can be a regex or glob to allow batch processing."""
-    pass
+    if pipe.returncode==0:
+        print(f'Audio extraction was successful for {path_to_video}')
+        output_path = pathlib.PurePath.joinpath(pathlib.Path(output_dir), pathlib.Path(audio_file))
```

Comment:
line 104 already constructed `Path(output_dir) / audio_file` to test if it `.exists()`. Better to assign it to a variable earlier and re-use that here.